packages/fast-components/src/slider/slider.ts

```diff
@@ -0,0 +1,327 @@
+
+    @observable
+    public track: HTMLDivElement;
+
+    @observable
+    public foregroundTrack: HTMLDivElement;
+
+    @observable
+    public thumb: HTMLDivElement;
+
+    @observable
+    public upperThumb: HTMLDivElement;
+
+    @observable
+    public lowerThumb: HTMLDivElement;
+
+    @observable
+    public direction: string = "ltr";
+
+    @observable
+    public isDragging: boolean = false;
+
+    @observable
+    public position: string;
+    @observable
+    public lowerPosition: string;
+    @observable
+    public upperPosition: string;
+    @observable
+    public trackWidth: number = 0;
+    @observable
+    public trackMinWidth: number = 0;
+
+    /**
+     * The element's value to be included in form submission when checked.
+     */
+    @attr
+    public value: string; // Map to proxy element.
+    private valueChanged(): void {
+        if (Number(this.value) === Number.NaN) {
+            this.value = "0";
+        }
+
+        if (this.proxy instanceof HTMLElement) {
+            this.updateForm();
+        }
+
+        const percentage: number =
+            this.direction !== "rtl"
+                ? (1 - (Number(this.value) / this.max - this.min)) * 100
+                : (Number(this.value) / this.max - this.min) * 100;
+
+        this.position = this.isDragging
+            ? `right: ${percentage}%; transition: all 0.1s ease;`
+            : `right: ${percentage}%; transition: all 0.2s ease;`;
+
+        this.$emit("change");
+    }
+
+    /**
+     * Min allowed value
+     */
+    @attr
+    public min: number = 0; // Map to proxy element.
+    private minChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.min = `${this.min}`;
+        }
+    }
+
+    /**
+     * Max allowed value
+     */
+    @attr
+    public max: number = 10; // Map to proxy element.
+    private maxChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.max = `${this.max}`;
+        }
+    }
+
+    @attr
+    public step: number = 1; // Map to proxy element.
+    private stepChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.step = `${this.step}`;
+        }
+    }
+
+    /**
+     * Orientation value, horizontal or vertical
+     */
+    @attr
+    public orientation: SliderOrientation = SliderOrientation.horizontal;
+
+    /**
+     * mode value, singleValue | dualThumb | adjustFromUpper | adjustFromLower
+     */
+    @attr
+    public mode: SliderMode = SliderMode.singleValue;
+
+    /**
+     * Set to true when the component has constructed
+     */
+    private constructed: boolean = false;
+    protected proxy = document.createElement("input");
+
+    constructor() {
+        super();
+
+        this.proxy.setAttribute("type", "range");
+        this.setAttribute("role", "slider");
+        this.setAttribute("tabindex", "0");
+        this.direction = this.getDirection();
+        this.constructed = true;
```

Comment:
depending on what you're using this for, you may not need it. Per @EisenbergEffect the $emit helper only runs after construction.

Reply:
Yep, I think `constructed` and checks can probably be removed now.

Reply:
I got rid of constructed and the setAttrs moved to the template